src/base-model.ts

```diff
@@ -371,13 +414,13 @@ export default abstract class Model<T> {
    * @param options: Additional options supported by AWS document client.
    * @returns  The scanned items (in the 1MB single scan operation limit) and the last evaluated key
    */
-  public scan(options?: Partial<ScanCommandInput>): Scan<T> {
+  public async scan(options?: Partial<ScanCommandInput>): Promise<Scan<T>> {
```

Comment:
For backward compatibility reason we do not want to change public API and make `scan` and `query` method async.

Solution here is to leave TableName empty and resolve it when async method `count`, `exec` or `execAll` are called.